fix CRM模块-（线索+客户）增加字段

fishSteak opened the pull request:
- 1、CRM模块增加了属性字段。公司介绍 官网 客户标签列表 国家列表

fishSteak committed:
fix crm添加了5个字段、增加验证字典还没有判断。

fishSteak committed:
fix  修改命名

fishSteak committed:
fix  修改

fishSteak committed:
fix  修改

c-tao commented on yudao-module-crm/yudao-module-crm-biz/src/main/java/cn/iocoder/yudao/module/crm/dal/dataobject/customer/CrmCustomerDO.java:
```diff
@@ -124,4 +127,40 @@ public class CrmCustomerDO extends BaseDO {
+     * <p>
+     * 存储公司名称，可能用于在 CRM 系统中展示和搜索
+     */
+    private String companyName;
```
是否与客户名称重复？（是一个东西吗）

fishSteak replied:
![image](https://github.com/user-attachments/assets/324fe9c5-9840-4e95-892c-3f46adbc4847)
不一致吧，丁哥说要的

fishSteak replied:
已删除

c-tao commented on yudao-module-crm/yudao-module-crm-biz/src/main/java/cn/iocoder/yudao/module/crm/dal/dataobject/customer/CrmCustomerDO.java:
```diff
@@ -124,4 +127,40 @@ public class CrmCustomerDO extends BaseDO {
+     * 公司介绍
+     * <p>
+     * 存储公司简介，通常用于展示公司背景信息
+     */
```
是否与备注重复？

c-tao commented on yudao-module-crm/yudao-module-crm-biz/src/main/java/cn/iocoder/yudao/module/crm/dal/dataobject/customer/CrmCustomerDO.java:
```diff
@@ -124,4 +127,40 @@ public class CrmCustomerDO extends BaseDO {
+    /**
+     * 官网
+     * <p>
+     * 存储公司官方网站的 URL 地址
```
同理，如果客户=公司的话，可以直接叫website或者url

c-tao commented on yudao-module-crm/yudao-module-crm-biz/src/main/java/cn/iocoder/yudao/module/crm/dal/dataobject/customer/CrmCustomerDO.java:
```diff
@@ -124,4 +127,40 @@ public class CrmCustomerDO extends BaseDO {
+     * 存储客户标签的 ID 列表，标签来自于字典，用于标记客户特征 crm_client_tag
+     */
+    @TableField(typeHandler = JacksonTypeHandler.class)
+    private List<Long> labelCodes;
```
为什么不直接叫label？

fishSteak replied:
因为是字典值、且有多个

fishSteak committed:
fix  CrmCustomer 增加5个查询入参

fishSteak committed:
fix  1、删除多选查询
原始代码有问题。例如更新线索的时候aop的spEL异常。
org.springframework.expression.spel.SpelEvaluationException: EL1007E: Property or field 'id' cannot be found on null

fishSteak commented:
原始代码有问题。例如更新线索的时候aop的spEL异常。
org.springframework.expression.spel.SpelEvaluationException: EL1007E: Property or field 'id' cannot be found on null

c-tao committed:
fix 完善了bo的字段。
2、修复线索转换客户时字段丢失。

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
fix 修改

fishSteak committed:
Merge branch 'somle-main' into somle-main-CRM(edit)

fishSteak commented:
> 原始代码有问题。例如更新线索的时候aop的spEL异常。 org.springframework.expression.spel.SpelEvaluationException: EL1007E: Property or field 'id' cannot be found on null

已修复

fishSteak committed:
fix 删除公司变量、完善跟进类型+客户来源分类字典。
2、修复SpelEvaluationException异常。

fishSteak commented:
根据最新的需求文档-已完成更改。

fishSteak commented:
线上表修改sql
[crm模块增加字段.txt](https://github.com/user-attachments/files/18817875/crm.txt)

fishSteak committed:
fix 修改公司介绍命名

fishSteak commented:
增加线上字段sql
[crm.txt](https://github.com/user-attachments/files/18820715/crm.txt)

fishSteak commented:
已完成更改。

fishSteak committed:
Merge remote-tracking branch 'origin/somle-main' into somle-main-CRM(edit)

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/src/main/java/cn/iocoder/yudao/module/erp/service/logistic/category/item/ErpCustomCategoryItemServiceImpl.java

fishSteak committed:
fix 修改

fishSteak committed:
Merge branch 'somle-main' into somle-main-CRM(edit)

# Conflicts:
#	yudao-module-erp/yudao-module-erp-biz/src/test/java/cn/iocoder/yudao/module/erp/service/logistic/category/ErpCustomCategoryServiceImplTest.java

fishSteak committed:
fix 修改

fishSteak committed:
fix 增加导入服务实现类

fishSteak commented:
> 原始代码有问题。例如更新线索的时候aop的spEL异常。 org.springframework.expression.spel.SpelEvaluationException: EL1007E: Property or field 'id' cannot be found on null

已修复

c-tao commented on yudao-module-crm/yudao-module-crm-api/src/main/java/cn/iocoder/yudao/module/crm/enums/DictTypeConstants.java:
```diff
@@ -15,5 +15,6 @@ public interface DictTypeConstants {
     String CRM_FOLLOW_UP_TYPE = "crm_follow_up_type"; // CRM 跟进方式
     String CRM_RECEIVABLE_RETURN_TYPE = "crm_receivable_return_type"; // CRM 回款方式
-
+    String CRM_CLIENT_TAG = "crm_client_tag"; // 客户标签
```
这里是否可以使用公用的

fishSteak replied:
用yudao-common包里面的？ 那个包没有。用的话，我替换过去？

fishSteak replied:
![image](https://github.com/user-attachments/assets/57308d60-8048-428e-9f68-2cce61d7f0ca)

fishSteak replied:
yudao对于模块内的 都是自己开辟了一个，关于XX模块的枚举类

fishSteak replied:
统一管理 确实方便一点，就是都要加载common包哦

fishSteak commented:
[Uploading 海关分类数据.txt…]()

fishSteak committed:
Revert "fix 增加导入服务实现类"

This reverts commit 125ec380299222caf6ad75bf3aea1487e1ab576f.

fishSteak committed:
Revert "fix 修改"

This reverts commit 8d05b397b20b2c35578acdd19b22c6b3c6b2d02a.

fishSteak committed:
Update DictTypeConstants.java

fishSteak committed:
fix 修改crm枚举名称

c-tao commented:
可以merge
